src/dwcahandler/dwca/core_dwca.py

```diff
@@ -84,19 +83,23 @@ def count_stat(self, content):
         """
         return len(content)
 
-    def _update_core_ids(self, core_df):
+    def _update_core_ids(self, core_df) -> str:
         """Generate core identifiers for a core data frame.
 
         UUID identifiers are generated for each row in the core data frame.
         These identifiers can be used to link core and extension records when no immediately
         useful identifier is available in the source data.
 
         :param core_df: The data frame to generate identifiers for
+        return id field
         """
-        if 'id' not in core_df.columns.to_list():
-            core_df.insert(0, 'id', core_df.apply(lambda _: uuid.uuid4(), axis=1), False)
+        id_field = "id"
+        if id_field not in core_df.columns.to_list():
+            core_df.insert(0, id_field, core_df.apply(lambda _: uuid.uuid4(), axis=1), False)
+            return id_field
         else:
-            core_df['id'] = core_df['id'].map(lambda _: uuid.uuid4())
+            raise ValueError("core df should not contain id column")
+        #    core_df['id'] = core_df['id'].map(lambda _: uuid.uuid4())
```

Comment:
Function name is a bit not showing what the function does,
Also making the id_field dynamically specified as a parameter

```suggestion
    def _generate_core_ids(self, core_df, id_field = "id") -> str:
        """Generate core identifiers for a core data frame.

        UUID identifiers are generated for each row in the core data frame.
        These identifiers can be used to link core and extension records when no immediately
        useful identifier is available in the source data.

        :param core_df: The data frame to generate identifiers for
        return id field
        """
        
        if id_field not in core_df.columns.to_list():
            core_df.insert(0, id_field, core_df.apply(lambda _: uuid.uuid4(), axis=1), False)
            return id_field
        else:
            raise ValueError("core df should not contain id column")
```